engine-nextgen/src/main/kotlin/graphql/nadel/validation/NadelHydrationValidation.kt

```diff
@@ -57,24 +58,29 @@ internal class NadelHydrationValidation(
                 errors.add(MissingHydrationActorService(parent, overallField, hydration))
                 continue
             }
-
-            val actorServiceQueryType = actorService.underlyingSchema.queryType
-            val actorField = actorServiceQueryType.getFieldAt(hydration.pathToActorField)
-            if (actorField == null) {
-                errors.add(MissingHydrationActorField(parent, overallField, hydration, actorServiceQueryType))
-                continue
+            if (newHydrationValidation) {
```

Comment:
you can simplify to something like
```kotlin
val schema = when {
  newHydrationValidation -> overallSchema
  else -> actorService.underlyingSchema
}

hydrationActorFieldExistsInSchema(hydration, errors, parent, overallField, actorService, isPolymorphicHydration, schema.queryType)
```

Reply:
Do you find this clearer over deciding the query type in the when?